sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/python/GpuPythonHelper.scala

```diff
@@ -29,13 +29,22 @@ import org.apache.spark.sql.internal.SQLConf
 
 object GpuPythonHelper extends Logging {
 
-  private val sparkConf = SparkEnv.get.conf
+  private lazy val sparkConf = SparkEnv.get.conf
   private lazy val rapidsConf = new RapidsConf(sparkConf)
   private lazy val gpuId = GpuDeviceManager.getDeviceId()
     .getOrElse(throw new IllegalStateException("No gpu id!"))
     .toString
   private lazy val isPythonPooledMemEnabled = rapidsConf.get(PYTHON_POOLED_MEM)
-    .getOrElse(rapidsConf.isPooledMemEnabled)
+    .getOrElse {
+      val rmmMode = GpuDeviceManager.rmmModeFromConf(rapidsConf)
+      if ((rmmMode & RmmAllocationMode.CUDA_ASYNC) != 0) {
+        logWarning("ASYNC pool is not supported by Rapids Python, fallback to normal pool")
+      } else if ((rmmMode & RmmAllocationMode.ARENA) != 0) {
+        logWarning("ARENA pool is not supported by Rapids Python, fallback to normal pool")
+      }
+      // Clear the uvm flag which is not relevant to memory pool
+      (rmmMode & (~RmmAllocationMode.CUDA_MANAGED_MEMORY)) != RmmAllocationMode.CUDA_DEFAULT
```

Comment:
Unless it is already, we should document it as well that the python worker supports only the regular pool mode. Could you please add some in the section for `spark.rapids.python.memory.gpu.pooling.enabled` in `advanced_configs.md`?